packages/api-plugin-promotions-discounts/src/discountTypes/order/applyOrderDiscountToCart.js

```diff
@@ -0,0 +1,94 @@
+import _ from "lodash";
+import formatMoney from "../../utils/formatMoney.js";
+import getEligibleItems from "../../utils/getEligibleItems.js";
+import getTotalEligibleItemsAmount from "../../utils/getTotalEligibleItemsAmount.js";
+import getTotalDiscountOnCart from "../../utils/getTotalDiscountOnCart.js";
+import recalculateCartItemSubtotal from "../../utils/recalculateCartItemSubtotal.js";
+
+/**
+ * @summary Map discount record to cart discount
+ * @param {Object} params - The action parameters
+ * @param {Array<Object>} discountedItems - The items that were discounted
+ * @param {Number} discountedAmount - The total amount discounted
+ * @returns {Object} Cart discount record
+ */
+export function createDiscountRecord(params, discountedItems, discountedAmount) {
+  const { promotion, actionParameters } = params;
+  const itemDiscount = {
+    promotionId: promotion._id,
+    discountType: actionParameters.discountType,
+    discountCalculationType: actionParameters.discountCalculationType,
+    discountValue: actionParameters.discountValue,
+    dateApplied: new Date(),
+    discountedItemType: "item",
+    discountedAmount,
+    discountedItems
+  };
+  return itemDiscount;
+}
+
+/**
+ * @summary Get the discount amount for a discount item
+ * @param {Object} context - The application context
+ * @param {Array<Object>} items - The cart to calculate the discount for
+ * @param {Object} discount - The discount to calculate the discount amount for
+ * @returns {Number} - The discount amount
+ */
+export function getCartDiscountAmount(context, items, discount) {
+  const merchandiseTotal = getTotalEligibleItemsAmount(items);
```

Comment:
Why do we calculate the merchandise total here, when the enhancer is doing the same?
Another question is if we calculate it here, why do we need the enhancer?